ext/OptimizationEnzymeExt.jl

```diff
@@ -13,30 +13,36 @@ function Optimization.instantiate_function(f::OptimizationFunction{true}, x,
 
```

Comment:
Since below you just assume that y is of size one, why not just return first(f.f(...)), and then use an active return value?

Reply:
Yes that should be possible. But would that work with the hessian implementation? I assumed based on https://enzyme.mit.edu//julia/dev/generated/autodiff/#Defining-a-function that it was being done purposefully and `autodiff` expected inplace function

Reply:
oh no it can return a value (and differentiate with respect to it). That was written like that to explicitly discuss a function that could return multiple values

Reply:
Got it

Reply:
Also fyi f.f is also being closured over, so you may want to have the first arg of this be f (which you can pass a first arg Const(f.f) to autodiff).